src/main/kotlin/com/demonwav/mcdev/platform/mixin/config/MixinConfigImportOptimizer.kt

```diff
@@ -0,0 +1,130 @@
+import com.intellij.json.psi.JsonElementGenerator
+import com.intellij.json.psi.JsonFile
+import com.intellij.json.psi.JsonObject
+import com.intellij.json.psi.JsonProperty
+import com.intellij.json.psi.JsonStringLiteral
+import com.intellij.lang.ImportOptimizer
+import com.intellij.openapi.util.EmptyRunnable
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+
+class MixinConfigImportOptimizer : ImportOptimizer {
+
+    /**
+     * Sorts a number of (qualified) class names based on their package names
+     * and the following rules:
+     *
+     *   - Each package name is compared separately
+     *   - Default packages come first
+     *   - Class names are only compared if in the same package
+     *
+     * Example:
+     *   ```
+     *   Main
+     *   com.demonwav.mcdev.Example
+     *   com.demonwav.mcdev.HelloWorld
+     *   com.demonwav.mcdev.asset.Assets
+     *   com.demonwav.mcdev.util.ActionData
+     *   ```
+     */
+    private object ClassPackageComparator : Comparator<String> {
+
+        // TODO: Consider implementing with Comparator.lexicographical() (in sorting.kt)
+        // Different here is that the class name is considered separately,
+        // so that default packages come first
+        override fun compare(class1: String, class2: String): Int {
+            val parts1 = class1.split('.')
+            val parts2 = class2.split('.')
+
+            val end = Math.min(parts1.size - 1, parts2.size - 1)
+            for (i in 0 until end) {
+                val result = parts1[i].compareTo(parts2[i], ignoreCase = true)
+                if (result != 0) {
+                    return result
+                }
+            }
+
+            if (parts1.size != parts2.size) {
+                // Default package always comes first
+                return Integer.compare(parts1.size, parts2.size)
+            }
+
+            // Compare class names
+            return parts1[end].compareTo(parts2[end], ignoreCase = true)
+        }
+    }
+
+    private val gson = Gson()
+
+    override fun supports(file: PsiFile) = file is JsonFile && file.fileType == MixinConfigFileType
+
+    override fun processFile(file: PsiFile): Runnable {
+        if (file !is JsonFile) {
+            return EmptyRunnable.getInstance()
+        }
+
+        val root = file.topLevelValue as? JsonObject ?: return EmptyRunnable.getInstance()
+
+        val mixins = processMixins(root.findProperty("mixins"))
+        val server = processMixins(root.findProperty("server"))
+        val client = processMixins(root.findProperty("client"))
+
+        if (mixins != null || server != null || client != null) {
+            return Task(file, mixins, server, client)
+        }
+
+        return EmptyRunnable.getInstance()
+    }
+
+    private fun processMixins(property: JsonProperty?): JsonArray? {
+        property ?: return null
+        val mixins = property.value as? JsonArray ?: return null
```

Comment:
Could just do `val mixins = property?.value as? JsonArray ?: return null` instead of the property check in it's own line.